syntax = "proto3";

package assign4;

service Database {
    rpc AccessDB (Request) returns (Response) {}
}

service Childnode {
    rpc TranslateChunk (ChunkRequest) returns (ChunkResponse) {}
    rpc HandleMiss (Request) returns (Response) {}
}

service Supernode {
    rpc TranslateChunk (ChunkRequest) returns (ChunkResponse) {}
    rpc HandleMiss (Request) returns (Response) {}
    rpc SendInfo (Info) returns (Confirm) {}
}

message ChunkRequest{
    bytes chunk=1;
}

message ChunkResponse{
    bytes chunk=1;
}

message Info{
    string ip=1;
}
message Confirm{
    bool checked=1;
}

message Request {
    string req = 1;
    bool from_super = 2;
}

message Response {
    string res = 1;
}
